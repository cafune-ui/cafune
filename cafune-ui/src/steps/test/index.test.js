import Steps from '../';
const { Step } = Steps;
import { render } from 'enzyme';
import { shallow, deep } from 'preact-render-spy';

const stepsArr = [
  { title: 'Step 1', desc: 'this is the step 1' },
  {
    title: 'Step 2',
    icons: { pend: <div>test</div>, finish: 'notice' },
    status: 'pend'
  },
  {
    title: 'Step 3',
    icons: { error: 'search', process: 'notice' },
    desc: 'now is the step 3',
    status: 'error'
  },
  { title: 'Step 4', desc: 'this is the step 4' },
  { title: 'Step 5', icons: {} }
];

const stepsArr2 = [
  { title: 'Step 1', desc: 'this is the step 1' },
  {
    title: 'Step 2'
  },
  {
    title: 'Step 2'
  },
  { title: 'Step 4', desc: 'this is the step 4' },
  { title: 'Step 5', icons: {} }
];

describe('Steps', () => {
  it('should render properly', () => {
    const wrapper = deep(
      <Steps step={1}>
        {stepsArr.map((item, ind) => (
          <Step {...item} key={ind} />
        ))}
      </Steps>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render with decimal ind', () => {
    const wrapper = deep(
      <Steps step={2} isDecimalInd>
        {stepsArr.map((item, ind) => (
          <Step {...item} key={ind} />
        ))}
      </Steps>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render in vertical', () => {
    const wrapper = deep(
      <Steps step={2} direction="vertical" icons={{ error: 'notice' }}>
        {stepsArr2.map((item, ind) => (
          <Step {...item} key={ind} />
        ))}
        <div>errpr</div>
      </Steps>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
