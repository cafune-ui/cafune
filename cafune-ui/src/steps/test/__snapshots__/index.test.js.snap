// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps should render in vertical 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "direction": "vertical",
      "icons": Object {
        "error": "notice",
      },
      "step": 2,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "desc": "this is the step 1",
          "key": 0,
          "title": "Step 1",
        },
        "children": Array [],
        "key": 0,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 1,
          "title": "Step 2",
        },
        "children": Array [],
        "key": 1,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "key": 2,
          "title": "Step 2",
        },
        "children": Array [],
        "key": 2,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "desc": "this is the step 4",
          "key": 3,
          "title": "Step 4",
        },
        "children": Array [],
        "key": 3,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "icons": Object {},
          "key": 4,
          "title": "Step 5",
        },
        "children": Array [],
        "key": 4,
        "nodeName": [Function],
      },
      VNode {
        "attributes": undefined,
        "children": Array [
          "errpr",
        ],
        "key": undefined,
        "nodeName": "div",
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Steps should render properly 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "step": 1,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "desc": "this is the step 1",
          "key": 0,
          "title": "Step 1",
        },
        "children": Array [],
        "key": 0,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "icons": Object {
            "finish": "notice",
            "pend": VNode {
              "attributes": undefined,
              "children": Array [
                "test",
              ],
              "key": undefined,
              "nodeName": "div",
            },
          },
          "key": 1,
          "status": "pend",
          "title": "Step 2",
        },
        "children": Array [],
        "key": 1,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "desc": "now is the step 3",
          "icons": Object {
            "error": "search",
            "process": "notice",
          },
          "key": 2,
          "status": "error",
          "title": "Step 3",
        },
        "children": Array [],
        "key": 2,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "desc": "this is the step 4",
          "key": 3,
          "title": "Step 4",
        },
        "children": Array [],
        "key": 3,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "icons": Object {},
          "key": 4,
          "title": "Step 5",
        },
        "children": Array [],
        "key": 4,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;

exports[`Steps should render with decimal ind 1`] = `
RenderContext {
  "0": VNode {
    "attributes": Object {
      "isDecimalInd": true,
      "step": 2,
    },
    "children": Array [
      VNode {
        "attributes": Object {
          "desc": "this is the step 1",
          "key": 0,
          "title": "Step 1",
        },
        "children": Array [],
        "key": 0,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "icons": Object {
            "finish": "notice",
            "pend": VNode {
              "attributes": undefined,
              "children": Array [
                "test",
              ],
              "key": undefined,
              "nodeName": "div",
            },
          },
          "key": 1,
          "status": "pend",
          "title": "Step 2",
        },
        "children": Array [],
        "key": 1,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "desc": "now is the step 3",
          "icons": Object {
            "error": "search",
            "process": "notice",
          },
          "key": 2,
          "status": "error",
          "title": "Step 3",
        },
        "children": Array [],
        "key": 2,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "desc": "this is the step 4",
          "key": 3,
          "title": "Step 4",
        },
        "children": Array [],
        "key": 3,
        "nodeName": [Function],
      },
      VNode {
        "attributes": Object {
          "icons": Object {},
          "key": 4,
          "title": "Step 5",
        },
        "children": Array [],
        "key": 4,
        "nodeName": [Function],
      },
    ],
    "key": undefined,
    "nodeName": [Function],
  },
  "length": 1,
}
`;
